httpchecker.go

```diff
@@ -180,7 +184,8 @@ func (c HTTPChecker) checkDown(resp *http.Response) error {
 // is specified on a HTTPChecker.
 var DefaultHTTPClient = &http.Client{
 	Transport: &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
This does not "enable checking HTTPS sites", this causes HTTPS checks to be ignored... which I think is undesirable for a health check. ;)

Reply:
Isn't it tlsCheckers purpose to ensure certs are correct? Using my case I need to be able to ignore the certs so I can test if the site is running (at least 1000 sites with different certs).  What would you think if i made it as an option to ignore the validation?

Reply:
Well, a HTTPS check requires _implicitly_ that TLS assets be correct, however a TLS checker's purpose is different. If you set this option, the HTTPS check will succeed even if browsers are not able to connect. Uh oh! (The TLS checker's purpose, on the other hand, does extra things like see if the certificate is expiring soon and if so, it emits a warning. HTTPS doesn't really care if the cert expires _soon_, only if it has expired.)

As the author of Caddy, I strongly, strongly, strongly _discourage_ ignoring validation. :wink: I personally will probably not accept a pull request that disables verification, at least on the HTTPS checker, since this undermines HTTPS entirely.

Reply:
Great work on Caddy. 

I'm trying to use CheckUp to request IP's using the host-header, I do this to ensure that the site works on all my different CDN's not just the first one.
`x509: cannot validate certificate for x.x.x.x because it doesn't contain any IP SANs` 

And as I can see in the Go implementations of [verify](https://golang.org/src/crypto/x509/verify.go) it will never use the host-header but only the IP as the host so it will not find a valid certificate.

What would you suggest to change for this, as I said I'm not a Go expert so if you could point me in the direction of a solution you would accept.

and btw Happy Holidays and happy New Year

Reply:
Thanks, and Happy Holidays to you too.

> I'm trying to use CheckUp to request IP's using the host-header

I don't understand what you mean by this; you are making a request that looks like this? `https://123.123.123.123/...` (i.e. HTTPS to some IP address) - if so, then the error is correct because the certificate is not valid for the given hostname. What _exactly_ are you trying to check with your healthcheck?

Reply:
I'm checking if a site is responding correctly based on a CDN setup. I don't know how to explain it better, so let me try with the configuration:
```
{
            "type": "http",
            "endpoint_name": "cdn02",
			"endpoint_url": "https://10.0.1.1",
			"attempts": 5,
			"headers":{"Host":["www.sitename.com"]}		
		},
}
```
That is why i also implemented the part with ensure that it set the host-header correctly the changes on line 90 and forward.

The point is i'm using the host-header to ensure that the webserver/cdn replies with the correct site. I know that the cert errror is correct based on the ip as host, but i need to use the ip to ensure that it actually tests all cdn locations and not the closest as it would do when using  https://www.sitename.com.

I know this is an edgy case, but i still think there is a problem in the header handling.

Reply:
I have removed the tlsconfig change